app/gui/src/controller/searcher.rs

```diff
@@ -514,12 +512,6 @@ impl Searcher {
     /// If `suggestion` is specified, the preview will contains code after applying it.
     /// Otherwise it will be just the current searcher input.
     pub fn preview(&self, suggestion: Option<&component::Suggestion>) -> FallibleResult {
-        let transaction_name = "Previewing Component Browser suggestion.";
-        let _skip = self
-            .graph
-            .undo_redo_repository()
-            .open_ignored_transaction_or_ignore_current(transaction_name);
-
```

Comment:
Now, we open the transaction once in `ComponentBrowserSearcher::setup_searcher`, and close it when the searcher presenter is dropped. When the user wants to discard the edit, we ignore the transaction (see `ComponentBrowserSearcher::abort_editing`). This way, we have a single transaction (I call it _Transaction_ in this comment) for the whole lifetime of the CB, and as far as I understand, that is what we need from the user perspective.

We can't use `open_ignored_transaction_or_ignore_current` here and in other methods of the searcher controller because it will ignore that _Transaction_ we opened in the presenter. Instead, we could use `let _skip = self.graph.get_or_open_transaction(transaction_name)`, and the `transaction_name` will be appended to the _Transaction_ log. I find it unnecessary, though.

If we remove the _Transaction_ from the presenter, we will have a bunch of individual transactions created by different searcher controller methods, like this one. Undoing node edit will then require undoing multiple stack frames, which is definitely not what we want.

Reply:
Shouldn't these actions create their own transactions? The order might be a little off, though. Like when you toggle visualization with the CB being opened, you will first discard the edit and only then revert the visualization change. 

I don't see how we can mitigate this. If we split CB actions to separate transactions it's not clear to me how merge them into a single undo frame.